examples/compiler/00_index.ipynb

```diff
@@ -0,0 +1,85 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# The Devito compiler\n",
+    "\n",
+    "The following series of notebooks will describe these compilation passes in a hands-on fashion. We will explore the heart of the `Operator` class, which orchestrates code generation and compilation. For more details, see the [Operator class](https://github.com/opesci/devito/blob/master/devito/operator.py).\n",
+    "\n",
+    "In Devito, an `Operator` carries out three fundamental tasks: generation of low-level code, JIT compilation, and execution. Overall, the lowering process performed by an `Operator` – from high-level equations to dynamically compiled and executable code – consists of multiple compiler passes, summarized in the Figure bellow. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/html": [
+       "\n",
+       "        <iframe\n",
+       "            width=\"800\"\n",
+       "            height=\"800\"\n",
+       "            src=\"https://ia601500.us.archive.org/13/items/compiler-stages/compiler-stages.pdf\"\n",
+       "            frameborder=\"0\"\n",
+       "            allowfullscreen\n",
+       "        ></iframe>\n",
+       "        "
+      ],
+      "text/plain": [
+       "<IPython.lib.display.IFrame at 0x7f50a8919320>"
+      ]
+     },
+     "execution_count": 1,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "from IPython.display import IFrame\n",
+    "IFrame(\"https://ia601500.us.archive.org/13/items/compiler-stages/compiler-stages.pdf\", width=800, height=800)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Many of the examples and exercises shown here are extracted from test cases found in the folder `devito/tests`, or from the documentation itself.\n",
+    "\n",
+    "## Outline\n",
```

Comment:
Could be a good time to write out the entire index, clearly with links to the available notebooks only.
The index, also, should/could have a tree structure. For example, to start with (we could revisit this in future PRs) we could have


* Preliminaries
    * [Data regions](https://github.com/opesci/devito/blob/master/examples/compiler/02_data_regions.ipynb)
    * Data dependence analysis
* Lowering
    * [Indexification](https://github.com/opesci/devito/blob/master/examples/compiler/01_lowering.ipynb)
    * Substitutions
    * Domain alignment
    * Eq -> LoweredEq
* Clustering
* Symbolic optimization via the Devito Symbolic Engine (DSE)
* Tree-fication
    * Clusters -> ScheduleTree
    * ScheduleTree -> Iteration/Expression Tree (IET)
    * IET examples
        * [Part A: top-down inspection](https://github.com/opesci/devito/blob/master/examples/compiler/03_iet-A.ipynb)
        * [Part B: bottom-up construction](https://github.com/opesci/devito/blob/master/examples/compiler/04_iet-B.ipynb)
* IET analysis
* Loop optimization via the Devito Loop Engine (DLE)
    * Classic transformations (SIMD, blocking, ...)
    * OpenMP parallelism
    * MPI parallelism
* Finalization     

which is rendered via
```
* Preliminaries
    * [Data regions](https://github.com/opesci/devito/blob/master/examples/compiler/02_data_regions.ipynb)
    * Data dependence analysis
* Lowering
    * [Indexification](https://github.com/opesci/devito/blob/master/examples/compiler/01_lowering.ipynb)
    * Substitutions
    * Domain alignment
    * Eq -> LoweredEq
* Clustering
* Symbolic optimization via the Devito Symbolic Engine (DSE)
* Tree-fication
    * Clusters -> ScheduleTree
    * ScheduleTree -> Iteration/Expression Tree (IET)
    * IET examples
        * [Part A: top-down inspection](https://github.com/opesci/devito/blob/master/examples/compiler/03_iet-A.ipynb)
        * [Part B: bottom-up construction](https://github.com/opesci/devito/blob/master/examples/compiler/04_iet-B.ipynb)
* IET analysis
* Loop optimization via the Devito Loop Engine (DLE)
    * Classic transformations (SIMD, blocking, ...)
    * OpenMP parallelism
    * MPI parallelism
* Finalization
```

yes, as you can see here I'm wondering whether the "data regions" notebook actually deserves a separate (preliminary) chapter...

Clearly this is just a suggestion ! What do you think?

Reply:
yes! that's what i've thought 
ofc this is way ahead lol ! 
thank you for making it more clear

Reply:
glad to help!